instrumentation/kafka-clients/kafka-clients-2.6/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/KafkaTracing.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.kafkaclients;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.kafka.KafkaConsumerRecordGetter;
+import io.opentelemetry.instrumentation.kafka.KafkaHeadersSetter;
+import java.util.concurrent.Future;
+import java.util.function.BiFunction;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.header.Headers;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class KafkaTracing {
+  private static final Logger logger = LoggerFactory.getLogger(KafkaTracing.class);
+
+  private static final TextMapGetter<ConsumerRecord<?, ?>> GETTER = new KafkaConsumerRecordGetter();
+
+  private static final TextMapSetter<Headers> SETTER = new KafkaHeadersSetter();
+
+  private final OpenTelemetry openTelemetry;
+  private final Instrumenter<ProducerRecord<?, ?>, Void> producerInstrumenter;
+  private final Instrumenter<ConsumerRecord<?, ?>, Void> consumerProcessInstrumenter;
+
+  KafkaTracing(
+      OpenTelemetry openTelemetry,
+      Instrumenter<ProducerRecord<?, ?>, Void> producerInstrumenter,
+      Instrumenter<ConsumerRecord<?, ?>, Void> consumerProcessInstrumenter) {
+    this.openTelemetry = openTelemetry;
+    this.producerInstrumenter = producerInstrumenter;
+    this.consumerProcessInstrumenter = consumerProcessInstrumenter;
+  }
+
+  /** Returns a new {@link KafkaTracing} configured with the given {@link OpenTelemetry}. */
+  public static KafkaTracing create(OpenTelemetry openTelemetry) {
+    return newBuilder(openTelemetry).build();
+  }
+
+  /** Returns a new {@link KafkaTracingBuilder} configured with the given {@link OpenTelemetry}. */
+  public static KafkaTracingBuilder newBuilder(OpenTelemetry openTelemetry) {
+    return new KafkaTracingBuilder(openTelemetry);
+  }
+
+  private TextMapPropagator propagator() {
+    return openTelemetry.getPropagators().getTextMapPropagator();
+  }
+
+  public <K, V> Producer<K, V> wrap(Producer<K, V> producer) {
```

Comment:
Can you add some javadocs to both `wrap()` methods? E.g. something like this
```suggestion
  /** Returns a decorated {@link Producer} that emits spans for each sent message. */
  public <K, V> Producer<K, V> wrap(Producer<K, V> producer) {
```